tools/install_ubuntu_dependencies.sh

```diff
@@ -99,11 +99,12 @@ function install_ubuntu_lts_latest_requirements() {
 # Install Ubuntu 20.04 packages
 function install_ubuntu_focal_requirements() {
   install_ubuntu_common_requirements
-
+  $SUDO add-apt-repository ppa:deadsnakes/ppa
```

Comment:
it's a bit hacky, so I want to split this out as a separate script and prompt the user to see if they want to install it. then with 24.04, we just remove this script entirely

Reply:
latest commit successfully builds DockerImage with ppa 3.11, i imagine tests should pass.

Prompt is similar to `extra_packages` flow. But If the user opts out to not wanting to use deadsnakes will it just do nothing? that's what its currently doing.